lib/std/Target.zig

```diff
@@ -2748,7 +2748,7 @@ pub fn is_libc_lib_name(target: std.Target, name: []const u8) bool {
             return true;
         if (eqlIgnoreCase(ignore_case, name, "resolv"))
             return true;
-        if (eqlIgnoreCase(ignore_case, name, "dl"))
+        if (eqlIgnoreCase(ignore_case, name, "dl") and !target.isWasm())
```

Comment:
With the libdl shim, isn't this wrong?

Reply:
That would pick the wrong `libdl`.

Reply:
I see, we have a special case for WASI here:

https://github.com/ziglang/zig/blob/c96f9a017a863e1f8cb610b7caba60ce93ab5616/src/main.zig#L3766-L3821

Still seems not good that `is_libc_lib_name()` returns false for these since all other parts of the codebase (like `std.Build`) will get the wrong answer.

Can't we just make `is_libc_lib_name()` do the right thing, and move the special case in `main.zig` so it happens before the `is_libc_lib_name()` check?

Reply:
I don't know if this is worth too much changes. The WASI `dl` library is just an awful hack to hopefully make some apps compile out of the box. But it only contains stub functions. None of them work.

Reply:
Well the above should apply to the other emulated libraries as well - e.g. `wasi-emulated-clocks`, which seems a fair bit more useful than the silly `dl` shim. `is_libc_lib_name()` will currently say that `wasi-emulated-clocks` isn't a libc library name, misleading `std.Build`. This is why I'm suggesting we just make `is_libc_lib_name()` do what it's supposed to do and then move the code block in `main.zig` up a bit. Then, going forward, adding new library names will work just like it does for the other libcs we provide.